src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java

```diff
@@ -411,6 +415,33 @@ private void publishSurveyLaunchedEvent(UUID caseId, String questionnaireId, Str
         .debug("SurveyLaunch event published");
   }
 
+  private void publishNewAddressReportedEvent(String caseId, AddressIndexAddressSplitDTO address) {
+    log.with("caseId", caseId).info("Generating NewAddressReported event");
+
+    CollectionCaseNewAddress newAddress =
+        caseDTOMapper.map(address, CollectionCaseNewAddress.class);
+    newAddress.setId(caseId);
+    newAddress.setSurvey("CENSUS");
+
+    // TO DO Derivation of addressLevel to be clarified. For now set to value covering majority of
```

Comment:
If you make the marker "TODO" rather than "TO DO" , or "FIXME" then IDEs will flag as an unfinished task.

Reply:
My bad. Done.